module/os/linux/zfs/vdev_disk.c

```diff
@@ -198,6 +200,13 @@ vdev_disk_open(vdev_t *v, uint64_t *psize, uint64_t *max_psize,
 			blkdev_put(bdev, mode | FMODE_EXCL);
 		}
 
+		/*
+		 * If we're expanding and have BLKPG_RESIZE_PARTITION, then
+		 * there is no need for the kernel to rescan the partition
+		 * table. This will already have been  updated in userland
+		 * and udev won't need to recreate the device links.
+		 */
+#ifndef BLKPG_RESIZE_PARTITION
```

Comment:
Is there a way to check if userland succeeded with the resize and do the right thing?  I'm concerned that older user space binaries won't make this check, and when running with newer kmods they will assume userland handled it.  While probably unlikely, it's also possible userland might fail the resize and it might be nice to optionally have another go at it here.